pkg/operator/staticpod/controller/revision/metrics.go

```diff
@@ -0,0 +1,30 @@
+package revision
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type prometheusMetricsProvider struct {
+	componentName string
+}
+
+func newPrometheusMetricsProvider(name string) prometheusMetricsProvider {
+	return prometheusMetricsProvider{componentName: name}
+}
+
+func (p prometheusMetricsProvider) NewRevisionMetrics() *prometheus.GaugeVec {
+	revisionMetrics := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: p.componentName,
+			Name:      "latestRevision",
+			Help:      fmt.Sprintf("Current %s operator revision.", p.componentName),
+		},
+		[]string{"observedGeneration"},
+	)
+	if err := prometheus.Register(revisionMetrics); err != nil {
```

Comment:
note that this must be a singleton in the process. If we call this from tests, we have to wrap that into some `sync.Once` or similar.